src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -498,6 +500,11 @@ Async.CombinedBuilder<FieldValueInfo> getAsyncFieldValueInfo(
         executionContext.getDataLoaderDispatcherStrategy().fieldFetched(executionContext, parameters, dataFetcher, fetchedObject);
         fetchCtx.onDispatched();
         fetchCtx.onFetchedValue(fetchedObject);
+        // if it's a subscription, leave any reactive objects alone
+        if (!executionContext.isSubscriptionOperation()) {
+            // possible convert reactive objects into CompletableFutures
+            fetchedObject = ReactiveSupport.fetchedObject(fetchedObject);
+        }
```

Comment:
This is the main code - if its NOT a Subscription, we turns the Pubisher into a CF